mlir/include/mlir/Dialect/Rock/IR/RockGemmWrapperInterface.td

```diff
@@ -33,7 +33,7 @@ def RockGemmWrapperInterface : OpInterface<"RockGemmWrapperInterface"> {
           Return the OpOperand that corresponds to the operand argument
           that corresponds to the output result of the operation.
         }],
-        /*retType=*/"OpOperand &",
+        /*retType=*/"OpOperand *",
```

Comment:
Can you just return an index? Returning pointers seems fragile.

Reply:
I could, but then all the clients would need the much more verbose invocation `&op->getOpOperand(cop.getOutArgument())`, which still summons up the pointer.

I'm not too concerned about the fragility. If you go peek at `LinalgStructuredInterface` in `mlir/include/mlir/Dialect/Linalg/IR/LinalgInterfaces.td` , which is used in linalg bufferization, they return `OpOperand *` or SmallVector<OpOperand *>` all over the place. 

I'm pretty sure that, for a given `Operation`, its `OpOperand`s are always in the same place unless someone mutates the op in place (which, for bufferization, could only happen in `bufferize()`)

Reply:
Yeah, still don't like it but won't hold it up.

Reply:
This is reasonable for the same reason passing `Value` by value is reasonable: in both cases, you're ultimately passing a pointer to something that's in a use-def list that's immutable. (OpOperand just doesn't have those nice wrappers around it)